cmd/azureBlobUpload.go

```diff
@@ -0,0 +1,129 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+// AzureContainerAPI is used to mock Azure containerClients in unit tests
+type AzureContainerAPI interface {
+	NewBlockBlobClient(blobName string) (*azblob.BlockBlobClient, error)
+}
+
+// NewBlockBlobClient creates a blockBlobClient from a containerClient
+func NewBlockBlobClient(blobName string, api AzureContainerAPI) (*azblob.BlockBlobClient, error) {
+	return api.NewBlockBlobClient(blobName)
+}
+
+// UploadFile uploads a file to an Azure Blob Storage
+// The function uses the UploadFile function from the Azure SDK
+// We introduce this 'wrapper' for mocking reasons
+func UploadFile(ctx context.Context, api *azblob.BlockBlobClient, file *os.File, o azblob.UploadOption) (*http.Response, error) {
+	return api.UploadFile(ctx, file, o)
+}
+
+// Struct to store Azure credentials from specified JSON string
+type azureCredentials struct {
+	SASToken    string `json:"sas_token"`
+	AccountName string `json:"account_name"`
+	Container   string `json:"container_name"`
+}
+
+func azureBlobUpload(config azureBlobUploadOptions, telemetryData *telemetry.CustomData) {
+	err := runAzureBlobUpload(&config)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAzureBlobUpload(config *azureBlobUploadOptions) error {
+	// Prepare Credentials
+	log.Entry().Infoln("Start reading Azure Credentials")
+	var creds azureCredentials
+
+	err := json.Unmarshal([]byte(config.JSONCredentialsAzure), &creds)
+	if err != nil {
+		log.Entry().WithError(err).Warnf("Could not read JSONCredentialsAzure")
```

Comment:
It's good if you wrap these errors and return them. Otherwise they are lost somewhere in the console log output. While the fatal **step** error will be printed **again** at the end of the pipeline. Thus making it easier for the user to find out what is going on. 
No need to print a warning that way. We all have learnt to ignore warnings anyway.

Reply:
I have now replaced all logging warnings with error wrappers.